openstack/sharedfilesystems/v2/shares/requests.go

```diff
@@ -379,3 +379,89 @@ func Update(client *gophercloud.ServiceClient, id string, opts UpdateOptsBuilder
 	})
 	return
 }
+
+// ShowMetadata retrieves metadata of the specified share. To extract the retrieved
+// metadata from the response, call the Extract method on the ShowMetadataResult.
+func ShowMetadata(client *gophercloud.ServiceClient, id string) (r ShowMetadataResult) {
+	_, r.Err = client.Get(showMetadataURL(client, id), &r.Body, nil)
+	return
+}
+
+// SetMetadataOpts contains options for setting share metadata.
+// For more information about these parameters, please, refer to the shared file systems API v2,
+// Share Metadata, Show share metadata documentation.
+type SetMetadataOpts struct {
+	Metadata map[string]string `json:"metadata"`
+}
+
+// ToSetMetadataMap assembles a request body based on the contents of an
+// SetMetadataOpts.
+func (opts SetMetadataOpts) ToSetMetadataMap() (map[string]interface{}, error) {
+	return gophercloud.BuildRequestBody(opts, "")
+}
+
+// SetMetadataOptsBuilder allows extensions to add additional parameters to the
+// SetMetadata request.
+type SetMetadataOptsBuilder interface {
+	ToSetMetadataMap() (map[string]interface{}, error)
+}
+
+// SetMetadata sets metadata of the specified share. To extract the updated
+// metadata from the response, call the Extract method on the SetMetadataResult.
```

Comment:
Worth calling out the difference between SetMetadata and UpdateMetadata, as a comment for future maintainers:

SetMetadata allows adding new metadata items or updating values of existing metadata items; however, UpdateMetadata will drop existing metadata and replace it with new metadata key=value pairs as specified.

Reply:
done in https://github.com/gophercloud/gophercloud/pull/1656/commits/5c226fec8324d00d2ecfeeb05cb247fa58e6a293